lib/rubygems/test_case.rb

```diff
@@ -722,14 +723,26 @@ def util_remove_gem(spec)
   end
 
   ##
-  # Removes all installed gems from +@gemhome+.
+  # Removes all installed gems from +Gem.path+.
 
   def util_clear_gems
-    FileUtils.rm_rf File.join(@gemhome, "gems") # TODO: use Gem::Dirs
-    FileUtils.mkdir File.join(@gemhome, "gems")
-    FileUtils.rm_rf File.join(@gemhome, "specifications")
-    FileUtils.mkdir File.join(@gemhome, "specifications")
-    Gem::Specification.reset
```

Comment:
should we keep calling `Gem::Specification.reset` ?

Reply:
First, I moved it to `util_clear_specs`, then I looked at it, and saw it called  `Gem.pre_reset_hooks` & ` Gem.post_reset_hooks`.  Given the unresolved issues with hooks, I then ran the tests locally, they passed, then here, and they passed.  Decided to leave as is and see what happens...

I don't think the stability issues are going to all disappear in a few commits.  I think your hook PR and this will get it manageable.

What would you like me to do?  Thanks, Greg

Reply:
Sounds fine, just wanted to make sure it was removed intentionally

Reply:
In hindsight, I should have left in it and commented it.  Does `[skip ci]` work here, or does it retest?